uui/components/filters/FilterPickerBody.tsx

```diff
@@ -23,13 +23,24 @@ export function FilterPickerBody<TItem, TId>(props: FilterPickerBodyProps<TItem,
         handleDataSourceValueChange,
     } = usePickerInput<TItem, TId, PickerInputProps<TItem, TId>>({ ...props, shouldShowBody });
 
+    const prevValue = usePrevious(props.value);
+    const prevOpened = usePrevious(props.isOpen);
+
     const renderItem = (item: TItem, rowProps: DataRowProps<TItem, TId>) => {
         return <PickerItem title={ getName(item) } size="36" { ...rowProps } />;
     };
 
+    React.useLayoutEffect(() => {
+        if (prevOpened === props.isOpen && props.isOpen 
+            && prevValue !== props.value && props.value !== props.emptyValue
+            && props.selectionMode === 'single'
```

Comment:
A body was closing previously in `single` selection mode on filter value selection.
On the other hand, if `clear` was executed, a body was not closed. To support such behavior, logic `props.value !== props.emptyValue` was added.

@AlekseyManetov, WDYT, is it correct, or body should be closed on both actions, filter value select and filter value clear?

Reply:
Yes, we shouldn't close picker the body by 'Clear' actions, because it's a common use case when user open picker -> clear value -> select another.